### PR TITLE
Move itest into their own module and create a separate build phase for them

### PR DESCRIPTION
**Pull request by jhunzik**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,8 +35,8 @@ For more information about spotless checks see
 [here](https://github.com/diffplug/spotless/tree/master/plugin-gradle#custom-rules).
 
 #### Tests
-* Tests are run automatically with `./gradlew build`.
-* To skip all tests, add `-x test`.
+* All tests are run automatically with `./gradlew build`.
+* To skip all tests, add `-x test` and `-x integrationTest`.
```
Comment (emmberk) [inline README.md]: ✏️ Does `-x test` skip unit tests and `-x integrationTest` skip itests? I recommend including the command for skipping all tests and the command for just skipping itests.

File: ingest/build.gradle [hunk 2]
```diff
@@ -9,10 +9,23 @@ plugins {
...
 }
 
+sourceSets {
+    integrationTest {
+        java {
+            compileClasspath += main.output + test.output
+            runtimeClasspath += main.output + test.output
+            srcDir file("src/itest/java")
+        }
+        resources.srcDir file("src/itest/resources")
```
Comment (emmberk) [inline ingest/build.gradle]: ✏️ consistently use `itest` or `integrationTest`
It looks like this in IntelliJ right now:
image

File: ingest/build.gradle [hunk 3]
```diff
@@ -21,7 +34,14 @@ dependencies {
...
     implementation(Dependencies.ingestAPI())
     implementation(Dependencies.transformAPI())
+
     testImplementation(Dependencies.springBootStarterTest())
+
+    integrationTestImplementation(Dependencies.ingestAPI())
+    integrationTestImplementation(Dependencies.transformAPI())
+    integrationTestImplementation(Dependencies.springBootStarterTest())
+    integrationTestImplementation(Dependencies.springBootStarterWeb())
+    integrationTestImplementation(Dependencies.reactiveStreams())
```
Comment (emmberk) [inline ingest/build.gradle]: ❓Shouldn't the itests inherit all of the dependencies from `implementation`?

File: ingest/build.gradle (hunk 2 above)
Comment (jhunzik) [reply ingest/build.gradle]: Both the module name and directory are named `itest`, not sure why IntelliJ is doing that. I don't have an issue with the longer name.

Comment (emmberk) [reply ingest/build.gradle]: I prefer itest

File: README.md (hunk 1 above)
Comment (jhunzik) [reply README.md]: Ah yeah. I clarified that and must have remove it in the readme. I think i can just have test depend on integrationTest so we get:
all tests: ./gradlew build
no itests: ./gradlew build -x test

File: ingest/build.gradle (hunk 3 above)
Comment (jhunzik) [reply ingest/build.gradle]: That's what I thought, but it seems like it's not the case

Comment (emmberk) [reply ingest/build.gradle]: I think we should figure out how to make that work. I could see an issue where we don't maintain the dependencies between implementation and itestImplementation, which means that we're running the itests with different dependencies.

Comment (jhunzik) [reply ingest/build.gradle]: This was a nightmare to look into, I think it should be a separate pr.

Comment (emmberk) [reply ingest/build.gradle]: I'm not okay merging until we figure this out. I think this change introduces a lot of risk.

Comment (emmberk) [reply ingest/build.gradle]: CC @lessarderic

File: build.gradle [hunk 4]
```diff
@@ -86,6 +87,14 @@ subprojects {
         }
     }
 
+    idea{
```
Comment (jhunzik) [inline build.gradle]: IntelliJ doesn't play nice with Gradle when it comes to custom sourceSets. The best I could come up with is using the idea plugin. If this isn't worth adding another plugin, I can remove it.

**Comment by emmberk** [comment]
😢

**Pull request closed without merging**